frontend/src/metabase/admin/people/components/GroupSelect/GroupSelect.tsx

```diff
@@ -38,35 +46,50 @@ export const GroupSelect = ({
       </PopoverWithTrigger>
     );
   }
-  const other = groups.filter(g => !isAdminGroup(g) && !isDefaultGroup(g));
+
   const adminGroup = groups.find(isAdminGroup);
   const defaultGroup = groups.find(isDefaultGroup);
   const topGroups = [defaultGroup, adminGroup].filter(g => g != null);
+  const groupsExceptDefaultAndAdmin = groups.filter(
+    g => !isAdminGroup(g) && !isDefaultGroup(g),
+  );
+
+  const sections =
```

Comment:
The things we do to avoid a `let` and array pushes…if too complicated below, please tell me.

Reply:
I think you could use the `isNotNull` check in your filter, which might be a bit more readable.

Personally, I'd probably prefer a `let` with a `switch/case`, or just putting that logic in it's own function.

Reply:
Thanks @npfitz.

I've extracted a function and it uses `isNotNull`, thanks for the suggestion.

Function could've been used an array push instead of filtering.

Thing is, the Type definition in the Select component is surprisingly complicated.
By doing it like we are here, the type is inferred.